File: CryptoHack/RSA/rsa_starter_4.go
package main

import (
	"fmt"
	"math/big"
)

func main() {
	e := big.NewInt(65537)
	p, _ := new(big.Int).SetString("857504083339712752489993810777", 10)
	q, _ := new(big.Int).SetString("1029224947942998075080348647219", 10)
	p.Sub(p, big.NewInt(1))
	q.Sub(q, big.NewInt(1))

	toient := new(big.Int).Mul(p, q)
	fmt.Println(e.ModInverse(e, toient))
}


// This Go script code calculates the modular inverse of e = 65537  modulo varphi(n) , where varphi(n) = (p-1) x (q-1) , and prints the result. The value d  is the modular inverse of e  modulo varphi(n) , used as the private key in RSA.

File: CryptoHack/RSA/RSAstarter4.py
from Cryptodome.Util.number import *

p = 857504083339712752489993810777
q = 1029224947942998075080348647219
e = 65537
phi_n = (p-1)*(q-1)

print(pow(e,-1,phi_n))

# This code calculates the modular inverse of e modulo phi(n) , where e is the public exponent and phi(n) is the totient of n , with n = p x q . The result is the private exponent d used in RSA encryption, ensuring that e x d equiv 1 pmod{phi(n)} .

